numpy/lib/tests/test_histograms.py

```diff
@@ -141,6 +141,25 @@ def test_f32_rounding(self):
         counts_hist, xedges, yedges = np.histogram2d(x, y, bins=100)
         assert_equal(counts_hist.sum(), 3.)
 
+    def test_bool_conversion(self):
+        # gh-12107
+        # Should raise an warning on booleans
+        with assert_raises(RuntimeWarning):
+            hist, edges = np.histogram([True, True, False])
+
+        # Reference integer histogram
+        a = np.array([1, 1, 0], dtype=np.uint8)
+        int_hist, int_edges = np.histogram(a)
+
+        # Ensure that the histograms are equivalent, need to suppress
+        # the warnings to get the actual outputs
+        sup = suppress_warnings()
+        sup.filter(RuntimeWarning)
+        with sup:
```

Comment:
It would be better to write this as
```
with suppress_warnings() as sup:
    sup.filter(RuntimeWarning)
```
to conform to the common usage.

Reply:
You can also check the warnings here.

Reply:
This would be better overall, if
```
# Reference integer histogram
￼a = np.array([1, 1, 0], dtype=np.uint8)
￼int_hist = np.histogram(a)

assert_warns(RuntimeWarning):
    hist = np.histogram([True, True, False])'
assert_array_equal(hist, int_hist)
```

Reply:
I'm happy to change this, I was just copying a pattern I found further down in the file: https://github.com/numpy/numpy/blob/7c5a3f4f9135af6ebe33dcd3242ef3e6db115cd6/numpy/lib/tests/test_histograms.py#L270

Reply:
Additionally (unless I'm doing something wrong), attempting to retrieve variables from the `assert_warns` call doesn't work:
```
    def test_bool_conversion(self):
        # gh-12107
        # Should raise an warning on booleans
        with assert_raises(RuntimeWarning):
            hist, edges = np.histogram([True, True, False])
>       print(hist)
E       UnboundLocalError: local variable 'hist' referenced before assignment
```